sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/_transport.py

```diff
@@ -696,6 +702,12 @@ def connect(self):
                 http_proxy_port=http_proxy_port,
                 http_proxy_auth=http_proxy_auth,
             )
+        except WebSocketAddressException as exc:
+            raise AuthenticationException(
+                ErrorCondition.SocketError, # TODO: ClientError?
```

Comment:
SocketError is fine here since the address not found is a SocketException.

Reply:
yeah I think socket error makes sense to me too

Reply:
Then I would be more inclined towards raising a ConnectionError versus an Auth error

Reply:
In that case, this would be a breaking change since for uamqp, we currently raise an Authentication error for an invalid custom endpoint address.

This one seems like AuthenticationException _could_ make sense, so I'm thinking that a breaking change here wouldn't be worth it. Thoughts? @annatisch

Reply:
Agree, for keeping the transition to pyamqp smooth its better to be consistent now and then be right later.

Reply:
ConnectionError probably _would_ have made more sense - but I'm inclined to not change the behaviour.
Though I do question the use of Condition.SocketError - is that also what we were using before?
Seems like that should be ClientError to me...

Reply:
It was previously a SocketError as well, and the previous error was caught and an AMQPConnectionError raised [[here](https://github.com/Azure/azure-sdk-for-python/blob/5013c2d8464a518ff4f2a6ca691f7913e1620c6e/sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/_connection.py#L190)].

Reply:
keeping old behavior including SocketError ErrorCondition.

Reply:
I was looking at uamqp and we never had this socketerror condition.... so I'm not really sure why it was introduced.
I'm confused as the link above is also pyamqp, not uamqp - or was that not the intended comparison?

Reply:
ah sorry, I meant before as in a previous version of pyamqp. Previously, it was caught in the _connection.py pyamqp file I believe. In uamqp, we didn't have an error condition at all as you mentioned.